geomstats/information_geometry/gamma.py

```diff
@@ -0,0 +1,1200 @@
+            Optional, default : 'geodesic_shooting'
+
+
+        Returns
+        -------
+        tangent_vec : array-like, shape=[..., dim]
+            Initial velocity of the geodesic starting at base_point and
+            reaching point at time 1.
+        """
+        try:
+            if method == "geodesic_shooting":
+                return super(GammaMetric, self).log(
+                    point, base_point, n_steps, step, max_iter, verbose, tol
+                )
+            if method == "ode_bvp":
+                return self._log_bvp(point, base_point, n_steps, jacobian=True)
+        except RuntimeWarning:
+            print(
+                "The points at which to compute the log might be too far away in "
+                "terms of Riemmanian distance."
+                " Trying with a higher n_steps may solve this issue."
+            )
+        except UserWarning:
+            print(
+                "The points at which to compute the log might be too far away in "
+                "terms of Riemmanian distance."
+                " Trying with a higher n_steps may solve this issue."
+            )
+
+    def geodesic(
+        self,
+        initial_point,
+        end_point=None,
+        initial_tangent_vec=None,
+        solver="geomstats",
+    ):
+        """Generate parameterized function for the geodesic curve.
+
+        Geodesic curve defined by either:
+        - an initial point and an initial tangent vector,
+        - an initial point and an end point.
+
+        Parameters
+        ----------
+        initial_point : array-like, shape=[..., dim]
+            Point on the manifold, initial point of the geodesic.
+        end_point : array-like, shape=[..., dim], optional
+            Point on the manifold, end point of the geodesic. If None,
+            an initial tangent vector must be given.
+        initial_tangent_vec : array-like, shape=[..., dim],
+            Tangent vector at base point, the initial speed of the geodesics.
+            Optional, default: None.
+            If None, an end point must be given and a logarithm is computed.
+        solver : string, {'geomstats', 'vp'}
+            Solver to compute the log map (self._geodesic_vp or Connection.geodesic)
+            Optional, default : 'geomstats'
+
+        Returns
+        -------
+        path : callable
+            Time parameterized geodesic curve. If a batch of initial
+            conditions is passed, the output array's first dimension
+            represents time, and the second corresponds to the different
+            initial conditions.
+        """
+        if solver == "geomstats":
+            return super(GammaMetric, self).geodesic(
+                initial_point, end_point, initial_tangent_vec
+            )
+        if solver == "vp":
+            return self.geodesic_vp(initial_point, end_point, initial_tangent_vec)
+
+    def _geodesic_ivp(self, initial_point, initial_tangent_vec, n_steps=N_STEPS):
```

Comment:
This function should be tested, see e.g. [here](https://github.com/geomstats/geomstats/blob/9bad7b8cc618fc0b9cc3dee13d01698a8f86bb11/tests/tests_geomstats/test_dirichlet.py#L159).